### PR TITLE
OLS-447: Unit test for checking provider type

### PR DESCRIPTION
**Pull request by tisnik**
## Description

Unit test for checking provider type

## Type of change

- [ ] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library
- [x] Unit tests

## Related Tickets & Documents

- Related Issue #[OLS-447](https://issues.redhat.com//browse/OLS-447)
- Closes #

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] PR has passed all pre-merge test jobs.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- Run `make test-unit` as usual

### CODE REVIEW

File: tests/unit/llms/providers/test_providers.py [hunk 1]
```diff
@@ -88,3 +89,25 @@ def load(self):
     )
 
     assert my_provider.params["provider-param"] == 3
+
+
+def test_llm_provider_params_order__no_provider_type():
```
[the remark this answers is not in the document]
Comment (tisnik) [reply tests/unit/llms/providers/test_providers.py]: I'm ok with it, I just followed Ondras naming conventions

**Comment by tisnik** [comment]
/tide/squash

File: tests/unit/llms/providers/test_providers.py [hunk 2]
```diff
@@ -88,3 +89,25 @@ def load(self):
...
+
+def test_llm_provider_params_order_no_provider_type():
+    """Test how missing provider type is handled."""
+    config.init_empty_config()
+    config.dev_config.llm_params = {"provider-param": 3}
+
+    class MyProvider(LLMProvider):
+        @property
+        def default_params(self):
+            return {"provider-param": 1}
```
Comment (bparees) [inline tests/unit/llms/providers/test_providers.py:102]: can you also add a default parameter that is not being overwritten and ensure that the default parameter appears in the provider params?  or do you already have that test elsewhere?

Comment (bparees) [reply tests/unit/llms/providers/test_providers.py:102]: forget to push? i don't see it

Comment (tisnik) [reply tests/unit/llms/providers/test_providers.py]: added+pushed @bparees

**Pull request merged**